### PR TITLE
generate distinct cert for use by client

### LINKED ISSUE #1930: Proposal: Separate cert should be generated for client auth
Working on #1918, I've noticed that minikube uses a single certificate `~/.minikube/apiserver.crt` as the serving cert for the k8s apiserver _and_ for the client (kubectl) to authenticate to the apiserver.

While this works fine and is not a danger of any sort for a local development environment (which is ostensibly what minikube is for), it's not a good representation oh how things are done "in real life."

I'd like to propose that a separate certificate should be generated specifically for the purpose of client authentication.

After #1918 is merged, the amount of work required for this is almost trivial and I'd be happy to take on this work myself.

### PR DESCRIPTION
**Pull request by krancour**
Fixes #1930

### CODE REVIEW

File: pkg/minikube/bootstrapper/certs.go [hunk 1]
```diff
@@ -49,17 +49,20 @@ func SetupCerts(cmd CommandRunner, k8s KubernetesConfig) error {
...
+	apiServerPrivatePath := filepath.Join(localPath, "apiserver.key")
+	clientPublicPath := filepath.Join(localPath, "client.crt")
+	clientPrivatePath := filepath.Join(localPath, "client.key")
 
 	proxyClientCACert := filepath.Join(localPath, "proxy-client-ca.crt")
 	proxyClientCAKey := filepath.Join(localPath, "proxy-client-ca.key")
 	proxyClientPublicPath := filepath.Join(localPath, "proxy-client.crt")
 	proxyClientPrivatePath := filepath.Join(localPath, "proxy-client.key")
 
 	if err := generateCerts(
```
Comment (r2d4) [inline pkg/minikube/bootstrapper/certs.go]: Do you mind making this into a struct instead?  The arguments here are getting a little unwieldy.

**Comment by r2d4** [comment]
@minikube-bot ok to test

File: pkg/minikube/bootstrapper/certs.go (hunk 1 above)
Comment (krancour) [reply pkg/minikube/bootstrapper/certs.go]: Yep... agree.

Comment (krancour) [reply pkg/minikube/bootstrapper/certs.go]: @r2d4, I refactored, but chose a different approach. I found that it wasn't really necessary to pass all those paths into `generateCerts(...)`. It is just as easy to generate the paths _within_ that function and then the nasty function signature goes away.

**Pull request merged**